### PR TITLE
fix(github): merge-commit-first on shallow clones; HEAD-on-base diff

### PR DESCRIPTION
**Pull request by gregmagolan**
Follow-up to #1086 fixing two bugs surfaced by live behavior on shallow CI checkouts.

**Bug 1 — PR runs hit the PR Files API despite merge-commit-first.** `_head_parents` was reading parents via `git log -1 --pretty=%P`, which honors the `.git/shallow` overlay and returns empty on every shallow checkout — even when HEAD is a merge commit. That silently disabled the merge-commit-first path on the common `actions/checkout` default (`fetch-depth: 1`). Switch to `git cat-file -p HEAD` and parse `^parent ` header lines: the raw commit object always carries the parent SHAs regardless of shallow state. To make the resulting HEAD^1 usable when the parent OBJECT also isn't local (shallow case), factor out `_ensure_local_sha` — `cat-file -e` then targeted `git fetch --depth=1 origin <sha>`.

**Bug 2 — `push` to base branch produced an empty diff.** `_resolve_merge_base` was accepting `git merge-base HEAD <base_ref>` results equal to HEAD's own SHA, which happens on push-to-base / squash-merge landings where HEAD IS the new tip of `origin/main`. `git diff HEAD..HEAD` is empty; lint/format saw zero changed files and quietly did nothing. Reject HEAD-equal merge-base results and fall through to HEAD^1, which works for both merge commits (PR/merge_queue) and single-parent commits on the base branch.

**WARN at the API fallback site.** When we still end up calling the PR Files API (e.g. user has a custom `ref:` override that makes HEAD a non-merge commit), surface a one-line WARN explaining the cost and recommending the default merge-commit ref.

**Helper rename.** `_read_github_event` → `read_github_event` and `_detect_merge_group_base_sha` → `detect_merge_group_base_sha` so the new test module can load them.

#### Suggested release notes

- `aspect lint` / `aspect format` / `aspect gazelle` now use a local `git diff HEAD^1..HEAD` on shallow PR checkouts (`actions/checkout` with `fetch-depth: 1`) instead of falling back to the GitHub PR Files API — reduces GitHub App rate-limit pressure on PR builds.
- `push` events to a base branch now correctly diff against `HEAD^1` instead of producing an empty change set when HEAD is the new tip of `origin/<base>` (previously silently skipped all files).
- New WARN when the PR Files API path is exercised, with guidance on the `actions/checkout` config that avoids it.

### Test plan

- New test cases added — `aspect dev test-github-detect` covers pull_request, merge_group, and push event shapes for `detect_github_pr`, `detect_merge_group_base_sha`, `detect_commit_sha`, and `read_github_event`, plus edge cases (missing event file, malformed JSON, `ASPECT_GITHUB_PR_NUMBER` override).
- Existing snapshot tests pass: `test-gitlab-detect`, `test-lint-template-snapshots`, `test-format-template-snapshots`.
- Manual: verified the cat-file approach on a real shallow clone (`git clone --depth=1 …`) — `git log --pretty=%P` returns empty while `git cat-file -p HEAD` shows the parent header; the bug-2 case reproduced on a depth-2 clone of the most recent push to `main` (`git merge-base HEAD origin/main` returned HEAD's own SHA, `git cat-file -p HEAD` revealed HEAD^1, and `git diff HEAD^1..HEAD` showed the expected landed change).

### CODE REVIEW

File: crates/aspect-cli/src/builtins/aspect/lib/github.axl [hunk 1]
```diff
@@ -542,22 +580,43 @@ def _resolve_merge_base(ctx, base_ref):
...
+        mb_sha = mb.stdout.strip()
+        if mb_sha and mb_sha != head_sha:
+            return mb_sha, "git merge-base HEAD %s" % base_ref
 
     parents = _head_parents(ctx)
-    if len(parents) >= 2:
-        return parents[0], "git rev-parse HEAD^1 (merge commit first parent)"
+    if len(parents) >= 1:
+        parent_sha = parents[0]
+        if _ensure_local_sha(ctx, parent_sha):
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply crates/aspect-cli/src/builtins/aspect/lib/github.axl]: Resolved
